src/mtt_allocator.c

```diff
@@ -181,19 +186,54 @@ void background_thread_fini(MTTAllocator *mtt_allocator)
 
     pthread_join(bg_thread->bg_thread, NULL);
     log_info("MTT background thread closed");
+}
+
+void background_thread_fini(MTTAllocator *mtt_allocator)
+{
+    BackgroundThread *bg_thread = &mtt_allocator->bgThread;
+
+    background_thread_stop(bg_thread);
 
     pebs_destroy(mtt_allocator, &bg_thread->pebs);
-    pthread_mutex_destroy(bg_thread_cond_mutex);
+    pthread_mutex_destroy(&bg_thread->bg_thread_cond_mutex);
     pthread_cond_destroy(&bg_thread->bg_thread_cond);
 }
 
+static void mtt_allocator_prefork(void)
+{
+    MTTAllocatorNode *cnode = root;
+    while (cnode) {
+        background_thread_stop(&cnode->allocator->bgThread);
+        pebs_disable(cnode->allocator, &cnode->allocator->bgThread.pebs);
+        cnode = cnode->next;
+    }
+}
+
+static void mtt_allocator_postfork(void)
+{
+    MTTAllocatorNode *cnode = root;
+    while (cnode) {
+        pebs_enable(cnode->allocator, &cnode->allocator->bgThread.pebs);
+        background_thread_start(&cnode->allocator->bgThread);
+        cnode = cnode->next;
+    }
+}
+
+static void mtt_atfork_init(void)
```

Comment:
nit: mtt_atfork_register_handlers ? Current name sounds a little bit like some initialization function that is performed during ```fork```, not a registration of *atfork* handlers, but maybe it's just my opinion...